.github/workflows/e2e-semgrep-ci.yml

```diff
@@ -1,4 +1,7 @@
+#TODO: The goal of this workflow is ???
```

Comment:
This workflow is set up to do some very basic E2E checks - makes sure that semgrep will still run both on a pull request and via `semgrep ci`. The context here is that we had some cases where releases went out that were broken in simple ways, and this gives us some confident that nightly semgrep still works.

Reply:
It also allows us to confirm fail-open behavior is still functioning as expected (e.g., we aren't passing on blocking findings)